receiver/prometheusreceiver/internal/transaction.go

```diff
@@ -169,11 +169,12 @@ func (tr *transaction) Commit() error {
 		return nil
 	}
 
-	ctx := obsreport.StartMetricsReceiveOp(tr.ctx, tr.receiverID, transport)
+	rec := obsreport.NewReceiver(obsreport.ReceiverSettings{ReceiverID: tr.receiverID, Transport: transport})
```

Comment:
This can be cached in the receiver struct.

Reply:
Done in latest commit